dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,40 +52,177 @@ public Pattern getParameterPattern() {
+                default:
+                    throw new SQLException("Unpexpected procedure parameter type for parameter " +
+                                           rs.getString("COLUMN_NAME"));
+            }
+        }
+    }
+
+    private class DatabaseFunction extends DatabaseObject {
+
+        private DatabaseFunction(DatabaseObjectName objName) {
+            super(objName);
+        }
+
+        @Override
+        protected ResultSet getObjectMetaData() throws SQLException {
+            return getDBMetaData().getFunctions(null, name.getSchemaName(), name.getObjectName());
+        }
+
+        @Override
+        public ResultSet getColumns() throws SQLException {
+            return getDBMetaData().getFunctionColumns(null, name.getSchemaName(), name.getObjectName(), "%");
+        }
+
+        @Override
+        protected Direction getParamDirection(ResultSet rs) throws SQLException {
+            int paramType = rs.getInt("COLUMN_TYPE");
+            switch (paramType) {
+                case DatabaseMetaData.functionColumnIn:
+                    return Direction.INPUT;
+                case DatabaseMetaData.functionReturn:
+                    return Direction.RETURN_VALUE;
+                default:
+                    throw new SQLException("Unpexpected function parameter type for parameter " +
+                                  rs.getString("COLUMN_NAME"));
+            }
+        }
+    }
+
+    private class DatabaseObjectName {
+        String schemaName, objectName;
+
+        private DatabaseObjectName(String defaultSchemaName, String objName) {
+            String[] qualifiers = objName.split("\\.");
+            schemaName = (qualifiers.length == 2) ? qualifiers[0] : defaultSchemaName;
+            objectName = (qualifiers.length == 2) ? qualifiers[1] : qualifiers[0];
+        }
+
+        private String getSchemaName() {
+            return schemaName;
+        }
+
+        String getObjectName() {
+            return objectName;
+        }
+    }
+
+    DatabaseObject findStoredRoutine(DatabaseObjectName objName) throws SQLException {
+        List<DatabaseObject> prioritisedCandidates = Arrays.asList(
+            new DatabaseProcedure(objName),
+            new DatabaseFunction(objName));
+        for (DatabaseObject object : prioritisedCandidates) {
+            if (object.exists()) {
+                return object;
+            }
+        }
+        throw new SQLException("Procedure/function " + objName.getObjectName() + " does not exist");
     }
 
-    public Map<String, DbParameterAccessor> getAllProcedureParameters(
-            String procName) throws SQLException {
-        throw new UnsupportedOperationException();
+    public Map<String, DbParameterAccessor> getAllProcedureParameters(String callableName)
+            throws SQLException {
+        String name = callableName.toUpperCase();
+        return findStoredRoutine(new DatabaseObjectName(getConnection().getSchema(), name)).getParams();
```

Comment:
Maybe it could be just `parseDatabaseObjectName` or even `parseObjectName` for now. Can be renamed some day if some day we have more options and risk of mis-interpretation. Possibly objeName.toUpperCase() can be injected here (to be honest I'm not sure about case sensitivity in Derby but if uppercase of name is needed, I guess it would be fine for schema name too).